easybuild/easyblocks/s/siesta.py

```diff
@@ -173,11 +176,60 @@ def configure_step(self):
             if netcdff_loc:
                 regex_subs.extend([
                     (r"^(LIBS\s*=.*)$", r"\1 $(NETCDF_LIBS)"),
-                    (r"^(FPPFLAGS\s*=.*)$", r"\1 -DCDF $(NETCDF_INCLUDE)"),
+                    (r"^(FPPFLAGS\s*:?=.*)$", r"\1 -DCDF $(NETCDF_INCLUDE)"),
                 ])
                 netcdf_lib_and_inc = "NETCDF_LIBS = -lnetcdff\nNETCDF_INCLUDE = -I%s/include" % netcdff_loc
                 regex_newlines.append((r"^(COMP_LIBS\s*=.*)$", r"\1\n%s" % netcdf_lib_and_inc))
 
+            xmlf90 = get_software_root('xmlf90')
+            if xmlf90:
+                regex_subs.append((r"^(XMLF90_ROOT\s*=).*$", r"\1%s" % xmlf90))
+
+            libpsml = get_software_root('libPSML')
+            if libpsml:
+                regex_subs.append((r"^(PSML_ROOT\s*=).*$.*", r"\1%s" % libpsml))
+
+            libgridxc = get_software_root('libGridXC')
+            if libgridxc:
+                regex_subs.append((r"^(GRIDXC_ROOT\s*=).*$", r"\1%s" % libgridxc))
+
+            libxc = get_software_root('libxc')
+            if libxc:
+                regex_subs.append((r"^#(LIBXC_ROOT\s*=).*$", r"\1 %s" % libxc))
+
+            elpa = get_software_root('ELPA')
+            if elpa:
+                elpa_ver = get_software_version('ELPA')
+                regex_subs.extend([
+                    (r"^(FPPFLAGS\s*:?=.*)$", r"\1 -DSIESTA__ELPA"),
+                    (r"^(FPPFLAGS\s*:?=.*)$", r"\1 -I%s/include/elpa-%s/modules" % (elpa, elpa_ver)),
+                    (r"^(LIBS\s*=.*)$", r"\1 -L%s/lib -lelpa" % elpa),
+                ])
+
+            elsi = get_software_root('ELSI')
+            if elsi:
+                elsi_libs = ['elsi', 'fortjson', 'OMM', 'MatrixSwitch', 'NTPoly']
+
+                if not elpa:
+                    elsi_libs.append('elpa')
+
+                if os.path.isfile(os.path.join(elsi, 'lib', 'libpexsi.a')):
+                    elsi_libs.extend(['pexsi', 'superlu_dist', 'ptscotchparmetis', 'ptscotch', 'ptscotcherr',
+                                      'scotchmetis', 'scotch', 'scotcherr'])
```

Comment:
Should probably check that superlu, scotch etc are actually part of the dependency chain before adding this.

Reply:
Even better would be if ELSI libs where made into shared libs so they can keep track of their own deps...

That would also remove the duplication of libraries below.

Reply:
good point, updated https://github.com/easybuilders/easybuild-easyblocks/pull/1883, will update this one